store/src/java/com/zimbra/cs/event/logger/FileEventLogHandler.java

```diff
@@ -0,0 +1,140 @@
+package com.zimbra.cs.event.logger;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+
+import com.zimbra.common.util.Log;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.event.Event;
+import com.zimbra.cs.event.Event.EventContextField;
+import com.zimbra.cs.event.Event.EventType;
+
+/**
+ * Event log handler that logs events to a log file in a structured manner.
+ * This lets us re-index events or migrate them to a different backend.
+ */
+public class FileEventLogHandler implements EventLogHandler {
+
+    private EventFormatter formatter;
+    private static final Log log = ZimbraLog.eventlog;
+
+    public FileEventLogHandler(EventFormatter formatter) {
+        this.formatter = formatter;
+    }
+
+    @Override
+    public void log(Event event) {
+        String logString = formatter.toLogString(event);
+        if (logString != null) {
+            log.info(logString);
+        }
+    }
+
+    @Override
+    public void shutdown() {
+        //nothing to do here
+    }
+
+    static interface EventFormatter {
+        String toLogString(Event event);
+        Event fromLogString(String string);
+    }
+
+    static class ZimbraEventFormatter implements EventFormatter {
+
+        private static final String KEY_TYPE = "type";
+        private static final String KEY_TIMESTAMP = "timestamp";
+        private static final String KEY_ACCT_ID = "id";
+
+        @Override
+        public String toLogString(Event event) {
+            String accountId = event.getAccountId();
+            String type = event.getEventType().toString();
+            String timestamp = String.valueOf(event.getTimestamp());
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, Object> eventMap = new HashMap<>();
```

Comment:
Thanks for teaching me something new!  I haven't noticed the diamond operator in `new HashMap<>()` before where `<>` on the RHS is a shorthand for what is between `<` and `>` on the LHS.  I think I have a new preferred way of initializing where generics are involved (-: !

Reply:
FYI: I did this in one of my PRs and was told not to at that time.  Do we have consensus then to move to this style for future updates?

Reply:
I certainly like this style better!

Reply:
I suspect it may have been me who complained - I probably didn't see the "<>" and assumed it was absent.

Reply:
I actually picked that up from @varunrisbud recently - I wasn't aware of that syntax either. Definitely saves keystrokes!